hail/src/main/scala/is/hail/expr/ir/Parser.scala

```diff
@@ -712,7 +712,7 @@ object IRParser {
         val cond = ir_value_expr(env)(it)
         val consq = ir_value_expr(env)(it)
         val altr = ir_value_expr(env)(it)
-        If.unify(cond, consq, altr)
+        If(cond, consq, altr)
```

Comment:
This previously silently promoted structs with different fields to have the same fields.

If with types

cond: bool
altr: array<struct{k: tuple(int32), left_index: int32}>
cnsq: array<struct{k: tuple(int32), left_index: int32, right_index: int32}>

would silently work because unify would choose one type and replace it with that of the other branch, with some (arbitrary? not certain of the significance order choice in If.unify on master ) precedence

Reply:
you're right that we shouldn't be doing this silently, but it's a capability that was never used (the cnsq/altr IRs coming from Python always had the same "python virtual type", vtype without requiredness)

Reply:
Yep, tests were unaffected by its presence. Left opportunities for bugs, but unrealized